docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+  # Connection information
+  Qos = "0" # Quality of Service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+```
+
+Additional configuration must be provided (in the service configuration.toml) to 
+- Trigger the collection of telemetry from the metrics cache and sending it into the appointed message bus.
+- Metrics will be published to an /edgex/telemetry/[service-name]/[metric-name] topic where the service name will be added per service and the metric name per metric (allowing subscribers to filter by service or metric name)
+
+``` yaml
+[[Telemetry]]
+Interval = "30s"
+TelemetryPublishTopicPrefix  = 'edgex/telemetry' # /<service-name>/<metric-name> will be added to this Publish Topic prefix
+```
+
+#### Library Support
+Each service will now need go-mod-messaging support.
+Each service would determine when and what metrics to collect and push to the message bus, but will use a common library choose for each EdgeX language supported (Go or C currently)
+
+There may be a desire to add common functionality in support of the REST handlers for getting the list of supported metrics and for enabling/disabling the metrics collection.  This can be explored at implementation time.
+
+Use of [go-metrics](https://github.com/rcrowley/go-metrics) (a GoLang library to publish application metrics) would allow EdgeX to utilize (versus construct) a library utilized by over 7 thousand projects.  It provides the means to capture various types of metrics in a registry (a sophisticated map).  The metrics can then be published (`reported`) to a number of well known systems such as InfluxDB, Graphite, DataDog, and Syslog.  go-metrics is a Go library made from original Java package https://github.com/dropwizard/metrics.
+
+A similar package would need to be selected for C.
+
+** Considerations in the use of go-metrics **
+- This is a Golang only library.  Using this library would not provide with any package to use for the C services.  If there are expectations for parity between the services, this may be more difficult to achieve given the features of go-metrics.
+- go-metrics will still require the EdgeX team to develop a bootstrapping apparatus to take the metrics configuration and register each of the metrics defined in the configuration in go-metrics.
+- go-metrics would also require the EdgeX team to develop the means to periodically extract the metrics data from the registry and ship it via message bus (something the current go-metrics library does not do).
+- While go-metrics offers the ability for data to be reported to other subsystems, it would required EdgeX to expose these capabilities (possibly through APIs) if a user wanted to export to these subsystems in addition to the message bus.

```

Comment:
I'm not sure what you mean by subsystems here? Can you clarify?

Reply:
@tonyespy , should be other systems, such as Prometheus or InfluxDB